components/event-sources/upgrade-job/process/patch_connectivity_validators.go

```diff
@@ -0,0 +1,94 @@
+package process
+
+import (
+	"encoding/json"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+
+	"github.com/pkg/errors"
+)
+
+var _ Step = &PatchConnectivityValidators{}
+
+const (
+	appReleaseNameKey = "meta.helm.sh/release-name"
+)
+
+type PatchConnectivityValidators struct {
+	name    string
+	process *Process
+}
+
+func NewPatchConnectivityValidators(p *Process) PatchConnectivityValidators {
+	return PatchConnectivityValidators{
+		name:    "Patch Application Connectivity Validators",
+		process: p,
+	}
+}
+
+func (s PatchConnectivityValidators) Do() error {
+	for _, validator := range s.process.State.ConnectivityValidators.Items {
+		patchedData, err := getDeploymentPatchData(validator)
+		if err != nil {
+			return err
+		}
+		_, err = s.process.Clients.Deployment.Patch(validator.Namespace, validator.Name, patchedData)
+		if err != nil {
+			return errors.Wrapf(err, "failed to patch validator deployment %s/%s", validator.Namespace, validator.Name)
+		}
+		s.process.Logger.Infof("Step: %s, patched connectivity validator: %s/%s deployment", s.ToString(), validator.Namespace, validator.Name)
+	}
+	return nil
+}
+
+func getNewContainerArgs(appName string) []string {
+	args := []string{
+		"/applicationconnectivityvalidator",
+		"--proxyPort=8081",
+		"--externalAPIPort=8080",
+		"--tenant=",
+		"--group=",
+		fmt.Sprintf("--eventServicePathPrefixV1=/%s/v1/events", appName),
+		fmt.Sprintf("--eventServicePathPrefixV2=/%s/v2/events", appName),
+		"--eventServiceHost=eventing-event-publisher-proxy.kyma-system",
+		"--eventMeshHost=eventing-event-publisher-proxy.kyma-system",
+		"--eventMeshDestinationPath=/publish",
+		fmt.Sprintf("--eventMeshPathPrefix=/%s/events", appName),
+		fmt.Sprintf("--appRegistryPathPrefix=/%s/v1/metadata", appName),
+		"--appRegistryHost=application-registry-external-api:8081",
+		fmt.Sprintf("--appName=%s", appName),
+		"--cacheExpirationMinutes=1",
+		"--cacheCleanupMinutes=2",
+	}
+	return args
+}
+
+func getDeploymentPatchData(oldDeployment appsv1.Deployment) ([]byte, error) {
+	appName := oldDeployment.Annotations[appReleaseNameKey]
+	var oldContainer corev1.Container
+	for _, c := range oldDeployment.Spec.Template.Spec.Containers {
+		oldContainer = c
```

Comment:
This will always evaluate to the last item in the `oldDeployment.Spec.Template.Spec.Containers` list, Could you please explain why is it like that?

Reply:
Right, and it is also true that there is only one container in this connectivity validator [deployment](https://github.com/kyma-project/kyma/blob/master/components/application-operator/charts/application/templates/validator.yaml#L34). Hence it works reliably.